### PR TITLE
Documentation for the Toffoli simulator

### PR DESCRIPTION
**Pull request by alan-geller**

### CODE REVIEW

File: articles/machines/toffoli-simulator.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ms.date: 1/16/20179
+ms.topic: article
+uid: microsoft.quantum.machines.toffoli-simulator
+---
+
+# Quantum Development Kit Toffoli Simulator
+
+The Quantum Development Kit provides a Toffoli simulator,
+which is a special-purpose simulator that can simulate quantum algorithms
+that are limited to X, CNOT, and multi-controlled X operations.
```
[the remark this answers is not in the document]
Comment (alan-geller) [reply articles/machines/toffoli-simulator.md]: GOod idea -- done.

File: articles/machines/index.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ Both are defined in the `Microsoft.Quantum.Simulation.Simulators` namespace.
 
 * A [full state vector simulator](xref:microsoft.quantum.machines.full-state-simulator), the `QuantumSimulator` class.
 * A [trace-based resource estimator](xref:microsoft.quantum.machines.qc-trace-simulator.intro), the `QCTraceSimulator` class.
+* A [Toffoli simulator](xref:microsoft.quantum.machines.toffoli-simulator), the `ToffoliSimulator` class.
```
Comment (tcNickolas) [inline articles/machines/index.md]: Do we need to add it to table of contents as well?

File: articles/machines/toffoli-simulator.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+it can simulate far more qubits.
+The Toffoli simulator can be used with millions of qubits, while the
+full state simulator is generally limited to about 30.
+
+This quantum simulator is exposed via the `ToffoliSimulator` class.
+To use the simulator, simply create an instance of this class and pass it
+to the `Run` method of the quantum operation you want to execute along
+with the rest of the parameters:
+
+```csharp
```
Comment (tcNickolas) [inline articles/machines/toffoli-simulator.md]: Is it IDisposable like the full state simulator? Does it support setting seed?

File: articles/machines/toffoli-simulator.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+# Quantum Development Kit Toffoli Simulator
+
+The Quantum Development Kit provides a Toffoli simulator,
+which is a special-purpose simulator that can simulate quantum algorithms
+that are limited to X, CNOT, and multi-controlled X quantum operations
+(all classical logic and computations are available).
+This simulator can be used to execute and debug a limited set of
+quantum algorithms written in Q# on your computer.
```
Comment (tcNickolas) [inline articles/machines/toffoli-simulator.md]: I would give an example of what kind of algorithms this is useful for - less of a cliffhanger but more useful :-)

File: articles/machines/toffoli-simulator.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+`ToffoliSimulator` constructor.
+Each additional qubit requires an additional byte of memory, so there is
+no significant cost to overestimating the number of qubits you'll need.
+
+For example:
+
+```csharp
+    var sim = new ToffoliSimulator(qubitCount: 1000000);
+    var res = myLargeOperation.Run(sim).Result;
+```
```
Comment (tcNickolas) [inline articles/machines/toffoli-simulator.md]: Where do we document the differences in DumpMachine and supported Asserts?

File: articles/machines/index.md (hunk 2 above)
Comment (alan-geller) [reply articles/machines/index.md]: Yes, I do -- thanks for the catch!!

File: articles/machines/toffoli-simulator.md (hunk 4 above)
Comment (alan-geller) [reply articles/machines/toffoli-simulator.md]: Done.

File: articles/machines/toffoli-simulator.md (hunk 5 above)
Comment (alan-geller) [reply articles/machines/toffoli-simulator.md]: Added.

**Comment by anpaz-msft** [comment]
Changed the target branch to `master`, so we can aggregate all changes for this release into a single branch.

**Pull request merged**